src/Service/ArchiveParser.php

```diff
@@ -0,0 +1,174 @@
+<?php
+
+namespace App\Service;
+
+use App\Entity\Event;
+use App\Repository\EventTypeRepository;
+use App\Repository\RegionRepository;
+use App\Repository\SettlementRepository;
+use App\Repository\SubgenreRepository;
+use App\Repository\WeaponRepository;
+use Symfony\Component\Console\Output\OutputInterface;
+use Doctrine\Common\Persistence\ObjectManager;
+
+class ArchiveParser
+{
+    private $typeRepository;
+    private $subgenreRepository;
+    private $settlementRepository;
+    private $weaponRepository;
+    private $regionRepository;
+    private $om;
+
+    public function __construct(EventTypeRepository $eventTypeRepository, RegionRepository $regionRepository, SubgenreRepository $subgenreRepository, SettlementRepository $settlementRepository, WeaponRepository $weaponRepository, ObjectManager $manager)
+    {
+        $this->typeRepository = $eventTypeRepository;
+        $this->subgenreRepository = $subgenreRepository;
+        $this->settlementRepository = $settlementRepository;
+        $this->weaponRepository = $weaponRepository;
+        $this->regionRepository = $regionRepository;
+        $this->om = $manager;
+    }
+
+    public function parseFile(OutputInterface $output, $filename)
+    {
+        $events = [];
+        $handle = @fopen($filename, "r");
+        if ($handle) {
```

Comment:
@daria-sieroshtan 
Instead of wrapping this gigantic stuff in "if ($handle)", better do early exit:
```PHP
if (not $handle) {
    throw \RuntimeException(sprintf("Cannot open input file %s for reading", $filename));
}
//more code, without offset
```